MAINTAINERS.md

```diff
@@ -1,4 +1,30 @@
-# Release process
+# Maintainers guide
```

Comment:
Personally, I think that only `## Release process` would be better in git because the content needs to be synced with the code. For the rest, I think that Notion is easier to keep up to date.